src/components/NanoContract/NanoContractTransactionHeader.component.js

```diff
@@ -0,0 +1,219 @@
+ * @param {Obejct} props.tx Transaction data
+ */
+export const NanoContractTransactionHeader = ({ tx }) => {
+  // TODO: use `toggleShrank` on 'onPress` of TouchableWithoutFeedback
+  const [isShrank, toggleShrank] = useState(false);
+
+  const isExpanded = () => !isShrank;
+
+  return (
+    <HathorHeader>
+      <HathorHeader.Central style={styles.headerCentral}>
+        <TouchableWithoutFeedback onPress={() => {}}>
+          <View style={styles.headerWrapper}>
+            <InfoContainer>
+              <TextValue title pb4>{getShortHash(tx.txId, 7)}</TextValue>
+              <TextLabel>{t`Transaction ID`}</TextLabel>
+            </InfoContainer>
+            {isShrank
+              && <HeaderShrank />}
+            {isExpanded()
+              && <HeaderExpanded tx={tx} />}
+          </View>
+        </TouchableWithoutFeedback>
+      </HathorHeader.Central>
+    </HathorHeader>
+  )
+};
+
+const HeaderShrank = () => (
+  <ArrowDownIcon />
+);
+
+/**
+ * It presents the expanded header of Nano Contract Transaction screen
+ * containing contextual information about the Nano Contract and the transaction.
+ *
+ * @param {Obejct} props
+ * @param {Obejct} props.tx Transaction data
+ */
+const HeaderExpanded = ({ tx }) => {
+  const baseExplorerUrl = useSelector((state) => state.networkSettings.explorerUrl);
+  const ncId = getShortHash(tx.ncId, 7);
+  const callerAddr = getShortContent(tx.caller, 7);
+  const hasFirstBlock = tx.firstBlock != null;
+
+  const navigatesToExplorer = () => {
+    const txUrl = `transaction/${tx.txId}`;
+    const explorerLink = combineURLs(baseExplorerUrl, txUrl);
+    Linking.openURL(explorerLink);
+  };
+
+  return (
+    <>
+      <View style={styles.wrapper}>
+        <InfoContainer>
+          <TransactionStatusLabel hasFirstBlock={hasFirstBlock} isVoided={tx.isVoided} />
+        </InfoContainer>
+        <InfoContainer>
+          <TextValue bold pb4>{ncId}</TextValue>
+          <TextLabel>{t`Nano Contract ID`}</TextLabel>
+        </InfoContainer>
+        <InfoContainer>
+          <TextValue bold pb4>{tx.ncMethod}</TextValue>
+          <TextLabel>{t`Blueprint Method`}</TextLabel>
+        </InfoContainer>
+        <InfoContainer>
+          <TextValue bold pb4>{getTimestampFormat(tx.timestamp)}</TextValue>
+          <TextLabel>{t`Date and Time`}</TextLabel>
+        </InfoContainer>
+        <InfoContainer lastElement>
+          <TextValue bold>{callerAddr}</TextValue>
+          {tx.isMine
+            && (
+            <View style={styles.headlineLabel}>
+              <Text style={styles.isMineLabel}>{t`From this wallet`}</Text>
+            </View>
+            )}
+          <TextLabel>{t`Caller`}</TextLabel>
+        </InfoContainer>
+        <TwoActionsWrapper>
+          <PrimaryTextButton title={t`See transaction details`} onPress={navigatesToExplorer} />
+        </TwoActionsWrapper>
+      </View>
+      {/* TODO: add <ArrowUpIcon /> when action list is ready to be showed */}
```

Comment:
Why is this a TODO? Will this PR be merged or wait the getActions method?

Reply:
It will be merged. But I will rip off this reference to actions. I believe it will be better.

Reply:
Done.